src/main/java/htsjdk/variant/variantcontext/VariantContext.java

```diff
@@ -665,10 +668,29 @@ public VariantContext subContextFromSample(String sampleName) {
      * @return the type of this VariantContext
      **/
     public Type getType() {
-        if ( type == null )
-            determineType();
+        return getType(false);
+    }
 
-        return type;
+    /**
+     * Determines (if necessary) and returns the type of this variation by examining the alleles it contains.
+     *
+     * @param ignoreNonRef If set to true, symbolic NON_REF alleles will not be considered for the type determination,
+     *                     which is required for handling GVCF files.
+     * @return the type of this VariantContext
+     **/
+    public Type getType(final boolean ignoreNonRef) {
```

Comment:
I have a slight preference for overloading the method instead of creating a different method, but for no particular reason, so I'm happy to implement whatever you suggest!

Reply:
This is fine.  I don't feel really feel strongly about it.